DigitalLearningSolutions.Data/Services/CustomPromptsService/CentreCustomPromptsService.cs

```diff
@@ -1,504 +1,284 @@
-        }
-
-        private static List<CustomPrompt> PopulateCustomPromptListFromCourseCustomPromptsResult(
-            CourseCustomPromptsResult? result
-        )
-        {
-            var list = new List<CustomPrompt>();
-
-            if (result == null)
-            {
-                return list;
-            }
-
-            var prompt1 = PopulateCustomPrompt(
-                1,
-                result.CustomField1Prompt,
-                result.CustomField1Options,
-                result.CustomField1Mandatory
-            );
-            if (prompt1 != null)
-            {
-                list.Add(prompt1);
-            }
-
-            var prompt2 = PopulateCustomPrompt(
-                2,
-                result.CustomField2Prompt,
-                result.CustomField2Options,
-                result.CustomField2Mandatory
-            );
-            if (prompt2 != null)
-            {
-                list.Add(prompt2);
-            }
-
-            var prompt3 = PopulateCustomPrompt(
-                3,
-                result.CustomField3Prompt,
-                result.CustomField3Options,
-                result.CustomField3Mandatory
-            );
-            if (prompt3 != null)
-            {
-                list.Add(prompt3);
-            }
-
-            return list;
-        }
-
-        private static List<CustomPromptWithAnswer> PopulateCustomPromptWithAnswerListFromCourseCustomPromptsResult(
-            CourseCustomPromptsResult? result,
-            DelegateCourseInfo delegateCourseInfo
-        )
-        {
-            var list = new List<CustomPromptWithAnswer>();
-
-            if (result == null)
-            {
-                return list;
-            }
-
-            var prompt1 = PopulateCustomPromptWithAnswer(
-                1,
-                result.CustomField1Prompt,
-                result.CustomField1Options,
-                result.CustomField1Mandatory,
-                delegateCourseInfo.Answer1
-            );
-            if (prompt1 != null)
-            {
-                list.Add(prompt1);
-            }
-
-            var prompt2 = PopulateCustomPromptWithAnswer(
-                2,
-                result.CustomField2Prompt,
-                result.CustomField2Options,
-                result.CustomField2Mandatory,
-                delegateCourseInfo.Answer2
-            );
-            if (prompt2 != null)
-            {
-                list.Add(prompt2);
-            }
-
-            var prompt3 = PopulateCustomPromptWithAnswer(
-                3,
-                result.CustomField3Prompt,
-                result.CustomField3Options,
-                result.CustomField3Mandatory,
-                delegateCourseInfo.Answer3
-            );
-            if (prompt3 != null)
-            {
-                list.Add(prompt3);
-            }
-
-            return list;
-        }
-    }
-}
+﻿namespace DigitalLearningSolutions.Data.Services.CustomPromptsService
```

Comment:
The split of this service is probably OK, I wonder if this is a good candidate for a full split into 2 separate services  though? (Also the DataServices and tests). The Centre Registration Prompts and Admin Fields are definitely more distinct than the Delegate and Admin users.

Reply:
I've split them fully now, let me know how that looks